python/ray/train/mosaic/mosaic_checkpoint.py

```diff
@@ -0,0 +1,31 @@
+from ray.air.checkpoint import Checkpoint
+import torch
+
+
+class MosaicCheckpoint(Checkpoint):
+    """A subclass of AIR Checkpoint for Composer models.
+    When reporting checkpoint in the Composer model training loop, a native composer
+    checkpoint file is loaded and converted to a ``MosaicCheckpoint`` object, which is
+    then reported via Ray.
+    Loading the saved model via ``MosaicCheckpoint`` involves removing the ``module``
+    prefix added via PyTorch DDP.
+    """
+
+    def get_model(
+        self,
+        model: torch.nn.Module,
+        strict: bool = False,
+    ) -> torch.nn.Module:
+        """Retrieve the model stored in this checkpoint."""
+        checkpoint_dict = self.to_dict()
+        model_state_dict = checkpoint_dict["state"]["model"]
+        torch.nn.modules.utils.consume_prefix_in_state_dict_if_present(
+            model_state_dict, "module."
+        )
+        model.load_state_dict(model_state_dict, strict=strict)
+        return model
+
+    @classmethod
+    def from_directory(cls, path: str):
+        composer_state_dict = torch.load(path)
+        return super().from_dict(composer_state_dict)
```

Comment:
can we not just return `super().from_dir(...)`?

Reply:
I actually think we should add a `from_path` method as a thin wrapper around `from_dir` as Mosaic checkpoints are just a single file

Reply:
Addressed